src/main/java/net/dancier/chatdancer/controllers/ChatController.java

```diff
@@ -0,0 +1,90 @@
+package net.dancier.chatdancer.controllers;
+
+import lombok.RequiredArgsConstructor;
+import net.dancier.chatdancer.dtos.ChatResponseDTO;
+import net.dancier.chatdancer.dtos.CreateMessageDTO;
+import net.dancier.chatdancer.dtos.CreateNewChatRequestDTO;
+import net.dancier.chatdancer.models.Chat;
+import net.dancier.chatdancer.models.Message;
+import net.dancier.chatdancer.services.ChatService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController()
+@RequestMapping("/chats")
+@RequiredArgsConstructor
+public class ChatController {
+
+    public static Logger log = LoggerFactory.getLogger(ChatController.class);
+    private final ChatService chatService;
+
+    @GetMapping
+    public ResponseEntity<List<Chat>> getAllChats(@RequestParam UUID userId) {
+        return new ResponseEntity<>(chatService.getAllChatsForUser(userId), HttpStatus.OK);
+    }
+
+    @GetMapping("/{chatId}")
+    public ResponseEntity<ChatResponseDTO> getChatById(@PathVariable UUID chatId) {
+
+        Chat requestedChat = chatService.getChatById(chatId);
+
+        return new ResponseEntity<>(convertChatToDto(requestedChat), HttpStatus.OK);
+    }
+
+    @GetMapping("/{chatId}/messages")
+    public ResponseEntity<List<Message>> getAllMessagesForChat(@PathVariable UUID chatId) {
+
+        return new ResponseEntity<>(chatService.getAllMessagesForChat(chatId), HttpStatus.OK);
+    }
+    
+    @PostMapping
+    public ResponseEntity<ChatResponseDTO> createNewChat(@RequestBody CreateNewChatRequestDTO createNewChatRequestDTO) {
+        Chat chat = convertDtoToChat(createNewChatRequestDTO);
+
+        Chat createdChat = chatService.createNewChat(chat);
+
+        ChatResponseDTO chatResponseDTO = convertChatToDto(createdChat);
+
+        log.info("Creating Chat: " + chatResponseDTO);
+
+        return new ResponseEntity<>(chatResponseDTO, HttpStatus.CREATED);
+
+
+    }
+
+    @PostMapping("/{chatId}/messages")
+    ResponseEntity<Message> createNewMessage(@PathVariable UUID chatId, @RequestBody CreateMessageDTO createMessageDTO) {
+
+        Message newMessage = Message.builder().text(createMessageDTO.getText()).authorId(createMessageDTO.getAuthorId()).build();
```

Comment:
If you use a method to convert from a dto as for chats I would say you should use the same approach for every conversion, having a method convertDtoToMessage